### ISSUE TITLE
Wayland support dosn't work

### ISSUE THREAD

**Issue by Janik-Haag** [AgathaSorceress/url-eater#2]
When trying to run the application on wayland it just crashes.
Here is a full back trace with `RUST_BACKTRACE=1` and `RUST_BACKTRACE=full` enabled:
```
./result/bin/url-eater /nix/store/5dl42hsfzqj9hv6byjwvz0ddzldllc97-filters.kdl
Loaded with categories:
	• Spotify
Error: 
  × Could not initialize clipboard context: XCB - couldn't establish conection: SetupFailed(SetupFailed { status: 0, protocol_major_version: 11, protocol_minor_version: 0, length: 16, reason: [65, 117, 116, 104,
  │ 111, 114, 105, 122, 97, 116, 105, 111, 110, 32, 114, 101, 113, 117, 105, 114, 101, 100, 44, 32, 98, 117, 116, 32, 110, 111, 32, 97, 117, 116, 104, 111, 114, 105, 122, 97, 116, 105, 111, 110, 32, 112, 114,
  │ 111, 116, 111, 99, 111, 108, 32, 115, 112, 101, 99, 105, 102, 105, 101, 100, 10] })
```

**Comment by AgathaSorceress** [comment]
Know issue, I have tried almost every available rust clipboard crate but for some reason all of them behave like this on Wayland. In some cases I managed to get it to not crash (but it also didn't do anything), in others it almost worked but could not read the clipboard. I don't have enough knowledge on how Wayland clipboard works, so I'm not sure how to continue with debugging this, but if anyone has suggestions/ideas, I would appreciate them

**Comment by ghost** [comment]
For what it's worth, I run url-eater succesfully on GNOME X11 and GNOME Wayland sessions. So perhaps the breakage is related to wlroots-based Wayland only? I wish I could provide more helpful insights, but wanted to add this here in case it might trigger something for devs that are more familiar with all these different wayland implementations...

**Comment by tamara-schmitz** [comment]
> For what it's worth, I run url-eater succesfully on GNOME X11 and GNOME Wayland sessions.

Yes I can confirm this. url-eater runs just fine as an X11 application in Xwayland there.

**Comment by tamara-schmitz** [comment]
cli-clipboard is supposed to grab Wayland first if available. X11 is the fallback:
https://github.com/allie-wake-up/cli-clipboard/blob/545878acf67047c699a52c8f37a9dbb978c6c536/src/linux_clipboard.rs?#L15

So it works for us cause we have that fallback. Not cool tho that it just discard the error from the Wayland attempt and only outputs the X11 error.

**Comment by AgathaSorceress** [comment]
Fixed by 57f7985 (hopefully)
